### PR TITLE
FLW-2315: Network issues flow

### PR DESCRIPTION
**Pull request by DRadmir**

### CODE REVIEW

File: fearless/Common/Extension/UIKit/UIButton.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Foundation
+import UIKit
+
+extension UIButton {
+    func textAndImage(spacing: CGFloat) {
```
Comment (bnsports) [inline fearless/Common/Extension/UIKit/UIButton.swift:5]: set keyword missed. let's name it 'setTextAndImageSpacing(_ spacing: CGFloat)' or 'setContentSpacing(_ spacing: CGFloat)'

File: fearless/Common/Protocols/SheetAlertPresentable.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    let handler: (() -> Void)?
+}
+
+struct SheetAlertPresentableViewModel {
+    let title: String
+    let titleStyle: SheetAlertPresentableStyle
+    let subtitle: String?
+    let subtitleStyle: SheetAlertPresentableStyle?
+    let actions: [SheetAlertPresentableAction]
```
Comment (bnsports) [inline fearless/Common/Protocols/SheetAlertPresentable.swift:16]: let's add constructor with default styles to simplify usage

File: fearless/Common/Protocols/SheetAlertPresentable.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+extension SheetAlertPresentable {
+    func present(
+        viewModel: SheetAlertPresentableViewModel,
+        from view: ControllerBackedProtocol?
+    ) {
+        var currentController = view?.controller
+
+        if currentController == nil {
+            currentController = UIApplication.shared.delegate?.window??.rootViewController
```
Comment (bnsports) [inline fearless/Common/Protocols/SheetAlertPresentable.swift:34]: it looks like you wanted to present this modal universally. best way to do it is to create new UIWindow and present modal here (it's apple recommendation for presenting alerts)

File: fearless/Modules/WalletMainContainer/WalletMainContainerPresenter.swift [hunk 4]
```diff
@@ -59,11 +69,13 @@ extension WalletMainContainerPresenter: WalletMainContainerViewOutput {
     }
 
     func didTapOnQR() {
-        router.showScanQr(from: view)
+//        router.showScanQr(from: view)
+        ChainRegistryFacade.sharedRegistry.connected()
```
Comment (bnsports) [inline fearless/Modules/WalletMainContainer/WalletMainContainerPresenter.swift:73]: mock should be removed

File: fearless/Modules/WalletMainContainer/WalletMainContainerPresenter.swift [hunk 5]
```diff
@@ -59,11 +69,13 @@ extension WalletMainContainerPresenter: WalletMainContainerViewOutput {
...
     }
 
     func didTapSearch() {
-        router.showSearch(from: view)
+//        router.showSearch(from: view)
+        ChainRegistryFacade.sharedRegistry.connect()
```
Comment (bnsports) [inline fearless/Modules/WalletMainContainer/WalletMainContainerPresenter.swift:78]: mock should be removed

File: fearless/Common/Services/ChainRegistry/ChainRegistry.swift [hunk 6]
```diff
@@ -136,6 +142,31 @@ final class ChainRegistry {
 }
 
 extension ChainRegistry: ChainRegistryProtocol {
+    // TODO: - Removed after demo
```
Comment (bnsports) [inline fearless/Common/Services/ChainRegistry/ChainRegistry.swift:145]: not sure that we can merge this to main develop branch. maybe it's make sense to create demo branch

File: fearless/Common/Services/ChainRegistry/NetworkIssuesProviderProtocol.swift [hunk 7]
```diff
@@ -0,0 +1,92 @@
+import Foundation
+
+protocol NetworkIssuesCenterListener: AnyObject {
+    func chainsWithIssues(_ chains: [ChainModel])
```
Comment (bnsports) [inline fearless/Common/Services/ChainRegistry/NetworkIssuesProviderProtocol.swift:4]: handleChainsWithIssues

File: fearless/Common/ViewController/SheetAlertViewLayout.swift [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+    init(viewModel: SheetAlertPresentableViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        applyStyle(viewModel: viewModel)
+        setupLayout()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        bind(viewModel: viewModel)
```
Comment (bnsports) [inline fearless/Common/ViewController/SheetAlertViewLayout.swift:57]: why here?

File: fearless/Modules/ChainAssetList/ChainAssetListPresenter.swift [hunk 9]
```diff
@@ -86,6 +94,25 @@ extension ChainAssetListPresenter: ChainAssetListViewOutput {
     func didTapAction(actionType: SwipableCellButtonType, viewModel: ChainAccountBalanceCellViewModel) {
         moduleOutput?.didTapAction(actionType: actionType, viewModel: viewModel)
     }
+
+    func didTapOnIssueButton(viewModel: ChainAccountBalanceCellViewModel) {
+        let topUpAction = SheetAlertPresentableAction(
```
Comment (bnsports) [inline fearless/Modules/ChainAssetList/ChainAssetListPresenter.swift:99]: closeAction ?

File: fearless/Modules/ChainAssetList/ViewModel/ChainAssetListViewModelFactory.swift [hunk 10]
```diff
@@ -169,6 +174,13 @@ private extension ChainAssetListViewModelFactory {
         let containsChainAssets = chainAssets.filter {
             $0.asset.name == chainAsset.asset.name
         }
+        var isNetworkIssues = false
+        containsChainAssets.forEach {
```
Comment (bnsports) [inline fearless/Modules/ChainAssetList/ViewModel/ChainAssetListViewModelFactory.swift:178]: containsChainAssets.first(where: {chainsWithIssues.contains($0.chain) }) != nil
one line

File: fearless/Modules/WalletMainContainer/WalletMainContainerViewLayout.swift [hunk 11]
```diff
@@ -72,19 +73,26 @@ final class WalletMainContainerViewLayout: UIView {
...
 
     // MARK: - FWSegmentedControl
 
-    let segmentedControl: FWSegmentedControl = {
-        let segment = FWSegmentedControl()
-        let items = ["Currencies", "NFTs"]
-        segment.setSegmentItems(items)
-        return segment
-    }()
+    let segmentedControl = FWSegmentedControl()
```
Comment (bnsports) [inline fearless/Modules/WalletMainContainer/WalletMainContainerViewLayout.swift:95]: why? don't you miss titles ?

File: fearless/Modules/WalletMainContainer/WalletMainContainerInteractor.swift [hunk 12]
```diff
@@ -62,6 +67,30 @@ final class WalletMainContainerInteractor {
         operationQueue.addOperation(operation)
     }
 
+    private func searchMissingAccounts() {
```
Comment (bnsports) [inline fearless/Modules/WalletMainContainer/WalletMainContainerInteractor.swift:70]: fetchMissingAccounts

File: fearless/Modules/WalletMainContainer/ViewModel/WalletMainContainerViewModel.swift [hunk 13]
```diff
@@ -4,4 +4,5 @@ struct WalletMainContainerViewModel {
     let walletName: String
     let selectedChainName: String
     let address: String?
+    let isNetworkIssues: Bool
```
Comment (bnsports) [inline fearless/Modules/WalletMainContainer/ViewModel/WalletMainContainerViewModel.swift:7]: hasNetworkIssues

File: fearless/Modules/NetworkIssuesNotification/View/NetworkIssuesNotificationViewLayout.swift [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func applyLocalization() {
+        bottomCloseButton.imageWithTitleView?.title = R.string.localizable.commonClose(
+            preferredLanguages: locale.rLanguages
+        )
+        titleLabel.text = "Notification"
```
Comment (bnsports) [inline fearless/Modules/NetworkIssuesNotification/View/NetworkIssuesNotificationViewLayout.swift:82]: localization?

File: fearless/Modules/NetworkIssuesNotification/View/NetworkIssuesNotificationTableCell.swift [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+        button.triangularedView?.sideLength = 6
+        button.triangularedView?.shadowOpacity = 0
+        button.imageWithTitleView?.titleFont = .p3Paragraph
+        return button
+    }()
+
+    let warningButton: UIButton = {
+        let button = UIButton()
+        button.setImage(R.image.iconWarning(), for: .normal)
+//        button.contentVerticalAlignment = .fill
```
Comment (bnsports) [inline fearless/Modules/NetworkIssuesNotification/View/NetworkIssuesNotificationTableCell.swift:55]: commented code

File: fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+        router.presentAccountOptions(
+            from: view,
+            locale: selectedLocale,
+            options: options.compactMap { $0 },
+            uniqueChainModel: UniqueChainModel(
+                meta: wallet,
+                chain: chain
+            )
+        ) { [weak self] _ in
+//            self?.interactor.markUnused(chain: chain)
```
Comment (bnsports) [inline fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift:57]: commented code

File: fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+        }
+    }
+
+    private func showSheetAlert(for chain: ChainModel) {
+        let topUpAction = SheetAlertPresentableAction(
```
Comment (bnsports) [inline fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift:62]: closeAction

File: fearless/Modules/WalletMainContainer/WalletMainContainerViewLayout.swift (hunk 11 above)
Comment (DRadmir) [reply fearless/Modules/WalletMainContainer/WalletMainContainerViewLayout.swift:95]: localized title. Set on applyLocalization method

File: fearless/Modules/NetworkIssuesNotification/View/NetworkIssuesNotificationTableCell.swift (hunk 15 above)
Comment (DRadmir) [reply fearless/Modules/NetworkIssuesNotification/View/NetworkIssuesNotificationTableCell.swift:55]: Oops ;]

**Review by bnsports — APPROVED** [review]

**Pull request merged**